enterprise/kernel/src/test/java/org/neo4j/kernel/impl/store/format/highlimit/NodeRecordFormatTest.java

```diff
@@ -123,6 +123,32 @@ public void useVariableLengthFormatWhenPropertyReferenceTooBig() throws IOExcept
         verifySameReferences( source, target);
     }
 
+    @Test
+    public void useVariableLengthFormatWhenRecordSizeIsTooSmall() throws IOException
+    {
+        NodeRecord source = new NodeRecord( 1 );
+        NodeRecord target = new NodeRecord( 1 );
+        source.initialize( true, randomFixedReference(), true, randomFixedReference(), 0L );
+
+        writeReadRecord( source, target, NodeRecordFormat.FIXED_FORMAT_RECORD_SIZE - 1 );
+
+        assertFalse( "Record should use variable length reference if format record is too small.", target.isUseFixedReferences() );
+        verifySameReferences( source, target);
+    }
```

Comment:
Fixed reference format is a single unit format with fixed number of bytes that required to store particular record data. For example for node record format will require https://github.com/neo4j/neo4j/pull/7670/files#diff-fdb66a08a0f6801727d057d81faa869eR52 bytes at least in a single unit.
In cases when single unit is smaller then that - we should fall back into var length format.